lib/trino-parquet/src/main/java/io/trino/parquet/reader/PrimitiveColumnReader.java

```diff
@@ -102,6 +104,10 @@ public static PrimitiveColumnReader createReader(RichColumnDescriptor descriptor
                 if (descriptor.getPrimitiveType().getOriginalType() == OriginalType.TIMESTAMP_MILLIS) {
                     return new Int64TimestampMillisColumnReader(descriptor);
                 }
+                if (descriptor.getPrimitiveType().getLogicalTypeAnnotation() instanceof TimestampLogicalTypeAnnotation &&
```

Comment:
what is original type in this case? It is not obvious from reading this sequence of `if` s that we will not bail out earlier even if logical type annotation is present.

Reply:
There is no "OriginalType.TIMESTAMP_NANOS".